### ISSUE TITLE
Flutter / Dart support

### ISSUE THREAD

**Issue by futurepaul** [zed-industries/feedback#145]
**Is your feature request related to a problem? Please describe.**

No syntax highlighting or language server for Flutter.

**Describe the solution you'd like**

Syntax highlighting, `dart format`, go to definition, hover, autocomplete, problems, and suggested fixes (for imports especially!) would be enough for me to do Flutter work from Zed.

LSP info here:
https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/tool/lsp_spec/README.md

Stretch goals would be some sort of "snippets" feature (flutter has a lot of boilerplate for class definitions) and a "run" button to give those IDE vibes.

Personally I'm doing Rust + Flutter projects right now, so I don't know if there's anything extra hard about showing problems from two different LSPs at once.

**Comment by s1dc0des** [comment]
yes please

**Comment by widersky** [comment]
There is [Language Server for Dart](https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md) but Flutter support could be somewhat problematic without official Flutter's Team support I think.

**Comment by caleb654** [comment]
Yes! VS code is getting to be very slow working on large Flutter and Dart projects.

**Comment by murshid1988** [comment]
Will this be a milestone?

**Comment by hovsater** [comment]
Work have begun to enable Zed to be extended with other languages. Unfortunately, I don't have an estimate on when this will be available for the general public. Stay tuned. 🙂

**Comment by mikezamayias** [comment]
Are there any updates?
